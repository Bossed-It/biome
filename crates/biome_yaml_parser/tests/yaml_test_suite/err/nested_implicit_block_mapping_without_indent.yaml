g: h: 70
d: e: 50
   f: 60
