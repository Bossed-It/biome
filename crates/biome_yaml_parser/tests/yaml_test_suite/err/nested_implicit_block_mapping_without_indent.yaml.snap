---
source: crates/biome_yaml_parser/tests/spec_test.rs
expression: snapshot
---
## Input
```yaml
g: h: 70
d: e: 50
   f: 60

```

## AST

```
YamlRoot {
    documents: YamlDocumentList [
        YamlDocument {
            bom_token: missing (optional),
            directives: YamlDirectiveList [],
            dashdashdash_token: missing (optional),
            node: YamlBlockMapping {
                mapping_start_token: MAPPING_START@0..0 "" [] [],
                properties: missing (optional),
                entries: YamlBlockMapEntryList [
                    YamlBlockMapImplicitEntry {
                        key: YamlFlowYamlNode {
                            properties: missing (optional),
                            content: YamlPlainScalar {
                                value_token: PLAIN_LITERAL@0..1 "g" [] [],
                            },
                        },
                        value: YamlBlockMapImplicitValue {
                            colon_token: COLON@1..3 ":" [] [Whitespace(" ")],
                            value: YamlBlockMapping {
                                mapping_start_token: MAPPING_START@3..3 "" [] [],
                                properties: missing (optional),
                                entries: YamlBlockMapEntryList [
                                    YamlBlockMapImplicitEntry {
                                        key: YamlFlowYamlNode {
                                            properties: missing (optional),
                                            content: YamlPlainScalar {
                                                value_token: PLAIN_LITERAL@3..4 "h" [] [],
                                            },
                                        },
                                        value: YamlBlockMapImplicitValue {
                                            colon_token: COLON@4..6 ":" [] [Whitespace(" ")],
                                            value: YamlFlowInBlockNode {
                                                flow_start_token: FLOW_START@6..6 "" [] [],
                                                flow: YamlFlowYamlNode {
                                                    properties: missing (optional),
                                                    content: YamlPlainScalar {
                                                        value_token: PLAIN_LITERAL@6..8 "70" [] [],
                                                    },
                                                },
                                                flow_end_token: FLOW_END@8..8 "" [] [],
                                            },
                                        },
                                    },
                                ],
                                mapping_end_token: MAPPING_END@8..9 "" [Newline("\n")] [],
                            },
                        },
                    },
                    YamlBlockMapImplicitEntry {
                        key: YamlFlowYamlNode {
                            properties: missing (optional),
                            content: YamlPlainScalar {
                                value_token: PLAIN_LITERAL@9..10 "d" [] [],
                            },
                        },
                        value: YamlBlockMapImplicitValue {
                            colon_token: COLON@10..12 ":" [] [Whitespace(" ")],
                            value: YamlBlockMapping {
                                mapping_start_token: MAPPING_START@12..12 "" [] [],
                                properties: missing (optional),
                                entries: YamlBlockMapEntryList [
                                    YamlBlockMapImplicitEntry {
                                        key: YamlFlowYamlNode {
                                            properties: missing (optional),
                                            content: YamlPlainScalar {
                                                value_token: PLAIN_LITERAL@12..13 "e" [] [],
                                            },
                                        },
                                        value: YamlBlockMapImplicitValue {
                                            colon_token: COLON@13..15 ":" [] [Whitespace(" ")],
                                            value: YamlFlowInBlockNode {
                                                flow_start_token: FLOW_START@15..15 "" [] [],
                                                flow: YamlFlowYamlNode {
                                                    properties: missing (optional),
                                                    content: YamlPlainScalar {
                                                        value_token: PLAIN_LITERAL@15..17 "50" [] [],
                                                    },
                                                },
                                                flow_end_token: FLOW_END@17..17 "" [] [],
                                            },
                                        },
                                    },
                                    YamlBogusBlockMapEntry {
                                        items: [
                                            MAPPING_START@17..21 "" [Newline("\n   ")] [],
                                            PLAIN_LITERAL@21..22 "f" [] [],
                                            COLON@22..24 ":" [] [Whitespace(" ")],
                                            FLOW_START@24..24 "" [] [],
                                            PLAIN_LITERAL@24..26 "60" [] [],
                                            FLOW_END@26..26 "" [] [],
                                        ],
                                    },
                                ],
                                mapping_end_token: MAPPING_END@26..27 "" [Newline("\n")] [],
                            },
                        },
                    },
                ],
                mapping_end_token: MAPPING_END@27..27 "" [] [],
            },
            dotdotdot_token: missing (optional),
        },
        YamlBogus {
            items: [
                MAPPING_END@27..27 "" [] [],
            ],
        },
    ],
    eof_token: EOF@27..27 "" [] [],
}
```

## CST

```
0: YAML_ROOT@0..27
  0: YAML_DOCUMENT_LIST@0..27
    0: YAML_DOCUMENT@0..27
      0: (empty)
      1: YAML_DIRECTIVE_LIST@0..0
      2: (empty)
      3: YAML_BLOCK_MAPPING@0..27
        0: MAPPING_START@0..0 "" [] []
        1: (empty)
        2: YAML_BLOCK_MAP_ENTRY_LIST@0..27
          0: YAML_BLOCK_MAP_IMPLICIT_ENTRY@0..9
            0: YAML_FLOW_YAML_NODE@0..1
              0: (empty)
              1: YAML_PLAIN_SCALAR@0..1
                0: PLAIN_LITERAL@0..1 "g" [] []
            1: YAML_BLOCK_MAP_IMPLICIT_VALUE@1..9
              0: COLON@1..3 ":" [] [Whitespace(" ")]
              1: YAML_BLOCK_MAPPING@3..9
                0: MAPPING_START@3..3 "" [] []
                1: (empty)
                2: YAML_BLOCK_MAP_ENTRY_LIST@3..8
                  0: YAML_BLOCK_MAP_IMPLICIT_ENTRY@3..8
                    0: YAML_FLOW_YAML_NODE@3..4
                      0: (empty)
                      1: YAML_PLAIN_SCALAR@3..4
                        0: PLAIN_LITERAL@3..4 "h" [] []
                    1: YAML_BLOCK_MAP_IMPLICIT_VALUE@4..8
                      0: COLON@4..6 ":" [] [Whitespace(" ")]
                      1: YAML_FLOW_IN_BLOCK_NODE@6..8
                        0: FLOW_START@6..6 "" [] []
                        1: YAML_FLOW_YAML_NODE@6..8
                          0: (empty)
                          1: YAML_PLAIN_SCALAR@6..8
                            0: PLAIN_LITERAL@6..8 "70" [] []
                        2: FLOW_END@8..8 "" [] []
                3: MAPPING_END@8..9 "" [Newline("\n")] []
          1: YAML_BLOCK_MAP_IMPLICIT_ENTRY@9..27
            0: YAML_FLOW_YAML_NODE@9..10
              0: (empty)
              1: YAML_PLAIN_SCALAR@9..10
                0: PLAIN_LITERAL@9..10 "d" [] []
            1: YAML_BLOCK_MAP_IMPLICIT_VALUE@10..27
              0: COLON@10..12 ":" [] [Whitespace(" ")]
              1: YAML_BLOCK_MAPPING@12..27
                0: MAPPING_START@12..12 "" [] []
                1: (empty)
                2: YAML_BLOCK_MAP_ENTRY_LIST@12..26
                  0: YAML_BLOCK_MAP_IMPLICIT_ENTRY@12..17
                    0: YAML_FLOW_YAML_NODE@12..13
                      0: (empty)
                      1: YAML_PLAIN_SCALAR@12..13
                        0: PLAIN_LITERAL@12..13 "e" [] []
                    1: YAML_BLOCK_MAP_IMPLICIT_VALUE@13..17
                      0: COLON@13..15 ":" [] [Whitespace(" ")]
                      1: YAML_FLOW_IN_BLOCK_NODE@15..17
                        0: FLOW_START@15..15 "" [] []
                        1: YAML_FLOW_YAML_NODE@15..17
                          0: (empty)
                          1: YAML_PLAIN_SCALAR@15..17
                            0: PLAIN_LITERAL@15..17 "50" [] []
                        2: FLOW_END@17..17 "" [] []
                  1: YAML_BOGUS_BLOCK_MAP_ENTRY@17..26
                    0: MAPPING_START@17..21 "" [Newline("\n   ")] []
                    1: PLAIN_LITERAL@21..22 "f" [] []
                    2: COLON@22..24 ":" [] [Whitespace(" ")]
                    3: FLOW_START@24..24 "" [] []
                    4: PLAIN_LITERAL@24..26 "60" [] []
                    5: FLOW_END@26..26 "" [] []
                3: MAPPING_END@26..27 "" [Newline("\n")] []
        3: MAPPING_END@27..27 "" [] []
      4: (empty)
    1: YAML_BOGUS@27..27
      0: MAPPING_END@27..27 "" [] []
  1: EOF@27..27 "" [] []

```

## Diagnostics

```
nested_implicit_block_mapping_without_indent.yaml:3:4 parse ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  × Expected a mapping but instead found 'f: 60'.
  
    1 │ g: h: 70
    2 │ d: e: 50
  > 3 │    f: 60
      │    ^^^^^
    4 │ 
  
  i Expected a mapping here.
  
    1 │ g: h: 70
    2 │ d: e: 50
  > 3 │    f: 60
      │    ^^^^^
    4 │ 
  
nested_implicit_block_mapping_without_indent.yaml:4:1 parse ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  × Expected a mapping but instead found the end of the file.
  
    2 │ d: e: 50
    3 │    f: 60
  > 4 │ 
      │ 
  
  i Expected a mapping here.
  
    2 │ d: e: 50
    3 │    f: 60
  > 4 │ 
      │ 
  
```
